mne_sandbox/connectivity/cfc.py

```diff
@@ -121,17 +122,136 @@ def phase_amplitude_coupling(inst, f_phase, f_amp, ixs, pac_func='ozkurt',
+        If `events` is provided, it is the time (in seconds) to include after
+        each event index. If a list of floats is given, then PAC is calculated
+        for each pair of `tmin` and `tmax`. Defaults to `max(inst.n_times)`.
+    n_cycles_ph : float, int | array of floats, shape (n_bands_phase,)
+        The number of cycles to be included in the window for each band-pass
+        filter for phase. Defaults to 3.
+    n_cycles_am : float, int | array of floats, shape (n_bands_amp,)
+        The number of cycles to be included in the window for each band-pass
+        filter for amplitude. Defaults to 3.
+    scale_amp_func : None | function
+        If not None, will be called on each amplitude signal in order to scale
+        the values. Function must accept an N-D input and will operate on the
+        last dimension. E.g., `sklearn.preprocessing.scale`.
+        Defaults to no scaling.
+    n_jobs : int
+        Number of jobs to run in parallel. Defaults to 1.
+    """
+    def __init__(self, f_phase, f_amp, ixs, sfreq, times,
+                 pac_func='ozkurt', tmin=0., tmax=1., n_cycles_ph=3,
+                 n_cycles_am=3, scale_amp_func=None, n_jobs=1):
+        events, times, tmin, tmax = _check_pac_times(None, times, tmin, tmax)
+        (ixs, f_phase, f_amp, n_cycles_ph,
+         n_cycles_am, pac_func) = _check_pac_data(
+            ixs, f_phase, f_amp, n_cycles_ph, n_cycles_am, pac_func)
+
+        self.f_phase = f_phase
+        self.f_amp = f_amp
+        self.tmin = tmin
+        self.tmax = tmax
+        self.times = times
+        self.ixs = ixs
+        self.sfreq = sfreq
+        self.pac_func = pac_func
+        self.n_cycles_ph = n_cycles_ph
+        self.n_cycles_am = n_cycles_am
+        self.scale_amp_func = scale_amp_func
+        self.n_jobs = n_jobs
+
+    def fit(self, X=None, y=None):
+        """
+        Parameters
+        ----------
+        X : None | array
+        y : None | array
+        """
+        return self
+
+    def transform(self, X, y=None):
+        """Estimate PAC values.
+
+        Parameters
+        ----------
+        X : array, shape (n_channels, n_times) |
+                (n_epochs, n_channels, n_times)
+            The input data.
+        y : None | array
+        """
+        n_ep = X.shape[0]
+        n_f_pairs = len(self.f_amp) * len(self.f_phase)
+        n_t_wins = len(self.tmin)
+        n_ch_pairs = len(self.ixs)
+
+        pac = np.zeros([X.shape[0], n_f_pairs * n_t_wins * n_ch_pairs])
+        for ii, ep_X in enumerate(X):
+            # Output will be shape (n_epochs, n_ch_pair, n_fr_pair, n_ti_win)
+            i_pac = _phase_amplitude_coupling(
+                ep_X, self.sfreq, self.f_phase, self.f_amp, self.ixs,
+                self.pac_func, None, self.tmin, self.tmax, self.times,
+                self.n_cycles_ph, self.n_cycles_am, self.scale_amp_func)[0]
+            # Reshape PAC so it's 2D and first dimension is Epochs
```

Comment:
no, keep it n dimensional, sklearn only needs the first dimension to be n_samples, the rest can be pipe as ndimensional arrays